### DISCUSSION TITLE
Template-based approach go-live date?

### DISCUSSION THREAD

**Discussion by ar0r** [community/community#78836] [GitHub Education] [answered]
### Hi everyone,

The recent post about the change to how repositories will be created in GitHub Classroom say that the change would take effect "in January."   Do you have a specific go live date?  Our term starts on January 3rd and we need to hit the ground running.

Will students be able to create new repositories for classrooms that were set up without a template repository under the old model?

**Comment by RyanHecht** [comment, marked as answer]
Hi @ar0r,

While we currently do not have any more concrete of a ship date other than "in January," we will be rolling it out in a way that is respectful of the fact that some terms will have already begun!

> Will students be able to create new repositories for classrooms that were set up without a template repository under the old model?

Yes, students will still be able to accept assignments created under the old model. The old students' repositories will be instantiations of the starter template repository, while the new students' repositories will be forks of the starter code template repository (and the "template" setting will be toggled off to ensure student repos are _not_ template repos).


There's one important caveat: if you're using a starter code template repository that has its visibility set to "public," you will not be able to have your student assignment repositories set to to "private." We will not be automatically turning on the new forks feature for classrooms with existing assignments that have this visibility mismatch, and will provide more information about how those classrooms can transition to the new system next month.

**Reply by RyanHecht** [reply]
Update @ar0r: The public beta has begun! https://github.blog/changelog/2024-01-22-migrating-github-classroom-assignment-repository-creation-from-create-from-template-to-forks-public-beta/
